source/link-controllers.test.ts

```diff
@@ -22,9 +21,31 @@ it('should abort all controllers when one is already aborted', () => {
 	const controller1 = new AbortController();
 	const controller2 = new AbortController();
 	const controller3 = new AbortController();
-	controller2.abort();
+
+	const reason = new Error('bad soup');
+	controller2.abort(reason);
 	linkControllers(controller1, controller2, controller3);
 	expect(controller1.signal.aborted).toBe(true);
 	expect(controller2.signal.aborted).toBe(true);
 	expect(controller3.signal.aborted).toBe(true);
+	expect(controller1.signal.reason).toBe(reason);
+	expect(controller2.signal.reason).toBe(reason);
+	expect(controller3.signal.reason).toBe(reason);
```

Comment:
The reason was not being shared here, the .abort method was called with its own not-yet-defined abort reason